From: "Barak A. Pearlmutter" <barak+git@cs.nuim.ie>
Date: Thu, 28 Jul 2011 14:16:13 +0100
Subject: tweak make arse

the install-arse target did not create the directories it installed
in, which messed up a parallel make.  It also built the script it
installed in the installation directory, rather than building it
locally and installing it.
---
 Makefile |   14 +++++++++-----
 1 files changed, 9 insertions(+), 5 deletions(-)

diff --git a/Makefile b/Makefile
index 432a728..44a79a8 100644
--- a/Makefile
+++ b/Makefile
@@ -155,15 +155,19 @@ arse-core.image: s9 contrib/arse.scm ext/unix.scm ext/curses.scm
 	$(BUILD_ENV) ./s9 !arse-core -n -l ext/unix.scm -l ext/curses.scm \
 		-l contrib/arse.scm -d arse-core.image
 
-install-arse: arse-core.image
-	cp arse-core.image $(DESTDIR)$(LIBDIR)
-	cp contrib/arse.help $(DESTDIR)$(LIBDIR)
+arse: prog/arse1.scm
 	sed -e "s|^#! /usr/local|#! $(prefix)|"	\
 	    -e "s|^#! \(.*\)/s9|#! \1/arse-core|"	\
 	    -e '/arse.scm"/d' \
-		<prog/arse1.scm >$(DESTDIR)$(BINDIR)/arse
+		<prog/arse1.scm >arse
+
+install-arse: arse arse-core.image
+	install -d -m 0755 $(DESTDIR)$(BINDIR)
+	install -d -m 0755 $(DESTDIR)$(LIBDIR)
+	install $C -m 0644 arse-core.image $(DESTDIR)$(LIBDIR)/
+	install $C -m 0644 contrib/arse.help $(DESTDIR)$(LIBDIR)/
+	install $C -m 0755 arse $(DESTDIR)$(BINDIR)/
 	ln -fs s9 $(DESTDIR)$(BINDIR)/arse-core
-	-chmod +x $(DESTDIR)$(BINDIR)/arse
 
 install-programs:
 	sed -e "s|^#! /usr/local|#! $(prefix)|"	\
-- 
